### PR TITLE
Explaining weights for linear regression models

### PR DESCRIPTION
**Pull request by lopuhin**
The "classifier" key is still used in `explain_weights` result even for regression models. Not sure if we need a better name - maybe "model" is more general?

### CODE REVIEW

**Comment by kmike** [comment]
scikit-learn uses 'estimator' name, what do you think about switching to it?

**Comment by lopuhin** [comment]
Yeah, I like it!

File: eli5/sklearn/explain_weights.py [hunk 1]
```diff
@@ -219,6 +235,90 @@ def explain_tree_feature_importance(clf, vec=None, top=_TOP, class_names=None,
...
+    Return an explanation of a linear regressor weights in the following
+    format::
+
+        {
+            "estimator": "<regressor repr>",
+            "method": "<interpretation method>",
+            "description": "<human readable description>",
+            "targets": [
+                {
+                    "targets <target name>,
```
Comment (kmike) [inline eli5/sklearn/explain_weights.py]: formatting looks off

File: eli5/sklearn/utils.py [hunk 2]
```diff
@@ -29,28 +37,49 @@ def get_feature_names(clf, vec=None, bias_name='<BIAS>', feature_names=None):
...
+
+
 def get_coef(clf, label_id):
     """
     Return a vector of coefficients for a given label,
     including bias feature.
     """
-    coef = clf.coef_[label_id]  # multiclass case, also works for binary
+    if len(clf.coef_.shape) == 2:
+        coef = clf.coef_[label_id]  # multiclass case, also works for binary
```
Comment (kmike) [inline eli5/sklearn/utils.py]: it seems the comment is incorrect if you needed this change

Comment (lopuhin) [inline eli5/sklearn/utils.py]: Right, SGDRegressor stores coefficients as a one-dim array, and binary does not make sense here, I'll fix comments

Comment (kmike) [inline eli5/sklearn/utils.py]: it seems it only worked for sparse `coef_` matrices, and tests used only such matrices

**Comment by kmike** [comment]
Looks good! Could you please also remove a TODO item in README?

File: eli5/sklearn/utils.py (hunk 2 above)
Comment (lopuhin) [inline eli5/sklearn/utils.py]: I think most (or all) classifiers we test have dense ``coef_`` matrices?

File: eli5/sklearn/explain_weights.py (hunk 1 above)
Comment (lopuhin) [inline eli5/sklearn/explain_weights.py]: Thanks for the catch! Fixed

**Comment by lopuhin** [comment]
I updated TODO, also mentioned that explaining predictions is currently not supported for regression models and trees (working on regression models at the moment).

**Comment by kmike** [comment]
Thanks, looks great!
